Add clusterId check when restarting nodes

HxpSerein opened the pull request:
## Description

Enhance the correctness check during node restart.

liyuheng55555 commented on iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/node/ClusterNodeStartUtils.java:
```diff
@@ -314,6 +315,25 @@ public static TSStatus confirmNodeRestart(
+          String.format(
+              "Reject %s restart. Because the clusterId of the current %s and the target cluster are inconsistent. "
+                  + "ClusterId of the current Node: %s, ClusterId of the target cluster: %s."
+                  + POSSIBLE_SOLUTIONS
+                  + "\t1. Change the cluster_id parameter in %s to match the target cluster.",
```
“Solution" might be not proper here, because we don't want user to modify system.properties. Maybe just suggest the user to check if the node configuration is correct?

HxpSerein replied:
Fixed

liyuheng55555 reviewed (changes requested):
I think there is no more need to pass cluster id in DN restart response now, please check 363 line of NodeManager and remove it.

OneSizeFitsQuorum reviewed (commented):
Missing confignode?

HxpSerein commented:
> Missing confignode?

At present, when the ConfigNode is restarted, it doesn't send an RPC to the ConfigNode leader. I've implemented persistent storage for the clusterId on the ConfigNode and added a TODO during the restart phase, laying the groundwork for further enhancements in the future.

liyuheng55555 commented on iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/node/NodeManager.java:
```diff
@@ -421,9 +421,11 @@ public TConfigNodeRegisterResp registerConfigNode(TConfigNodeRegisterReq req) {
         .setStatus(ClusterNodeStartUtils.ACCEPT_NODE_REGISTRATION)
-        .setConfigNodeId(nodeId);
+        .setConfigNodeId(nodeId)
+        .setClusterId(clusterId);
```
Not quite necessary here, CN's consensus layer will automatically sync cluster id

HxpSerein replied:
changed.

liyuheng55555 commented on iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:
```diff
@@ -449,7 +446,7 @@ private void storeRuntimeConfigurations(
 
     /* Store cluster ID */
-    IoTDBDescriptor.getInstance().getConfig().setClusterId(runtimeConfiguration.getClusterId());
+    storeClusterID(runtimeConfiguration.getClusterId());
```
After remove the cluster id from DN restart resp, NPE might happen here

HxpSerein replied:
fixed.

liyuheng55555 reviewed (changes requested):
PTAL~

liyuheng55555 commented:
> > Missing confignode?
> 
> At present, when the ConfigNode is restarted, it doesn't send an RPC to the ConfigNode leader. I've implemented persistent storage for the clusterId on the ConfigNode and added a TODO during the restart phase, laying the groundwork for further enhancements in the future.

Even if a wrong CN joining in accidently , I think it won't cause any serious problem. Remove it and add a new empty CN will fix everything. So for now it's fine to only verify DN's restart.

OneSizeFitsQuorum reviewed (approved):
LGTM

liyuheng55555 reviewed (approved):
LGTM. Those who add the wrong ConfigNode to the cluster, hope god bless them.
